### PR TITLE
Feature: (issue_1451) customized slice flag separator

### PR DESCRIPTION
**Pull request by FGYFFFF**
## What type of PR is this?
feature
_(REQUIRED)_

- bug
- cleanup
- documentation
- feature

## What this PR does / why we need it:
add  customized slice flag separator
_(REQUIRED)_

## Which issue(s) this PR fixes:
Feature https://github.com/urfave/cli/issues/1541
_(REQUIRED)_

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note
Feature: (https://github.com/urfave/cli/issues/1541)add  customized slice flag separator
```

### CODE REVIEW

**Comment by FGYFFFF** [comment]
The setting of "SliceFlagSeparator" is "global-level" because it is necessary to keep the same separator for each "sliceFlag". Therefore it is not set to "flag-level"

**Comment by FGYFFFF** [comment]
@dearchap Please help me, why does "CI" keep failing?&#x1F62D;

**Comment by dearchap** [comment]
Can you do a gofmt or goimports in main dir ?

**Comment by FGYFFFF** [comment]
> Can you do a gofmt or goimports in main dir ?

I did gofmt&goimports for the "flag_test.go" file, but it still failed.

**Comment by FGYFFFF** [comment]
@dearchap  Please help me review the code, I rely on this feature to do my work！💕💕💕

File: flag_test.go [hunk 1]
```diff
@@ -3384,3 +3384,30 @@ func TestSliceShortOptionHandle(t *testing.T) {
...
+		for idx, r := range ret {
+			if r != opts[idx] {
+				t.Fatalf("get %dth failed, wanted: %s, but get: %s", idx, opts[idx], r)
+			}
+		}
+		return nil
+	}
+
+	_ = app.Run(os.Args)
+	defaultSliceFlagSeparator = ","
```
Comment (dearchap) [inline flag_test.go:3412]: better to add the revert of defaultSliceFlagSeparator in a defer

File: flag_test.go [hunk 2]
```diff
@@ -3384,3 +3384,30 @@ func TestSliceShortOptionHandle(t *testing.T) {
...
+		"--stringSlice=" + strings.Join(opts, ";"),
+	}
+	app := NewApp()
+	app.Name = "multi_values"
+	app.Flags = []Flag{
+		&StringSliceFlag{Name: "stringSlice"},
+	}
+	app.SliceFlagSeparator = ";"
+	app.Action = func(ctx *Context) error {
+		ret := ctx.StringSlice("stringSlice")
```
Comment (dearchap) [inline flag_test.go:3402]: its better not to rely on Action here. If action doesnt get called the test would pass. why dont you set the destination filed of the string slice flag and then validate the slice contents

File: flag_test.go (hunk 1 above)
Comment (FGYFFFF) [reply flag_test.go]: done

File: flag_test.go (hunk 2 above)
Comment (FGYFFFF) [reply flag_test.go]: done! ✌️

**Review by dearchap — APPROVED** [review]

**Pull request merged**